src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/TestDriverComponent.scala

```diff
@@ -20,18 +26,302 @@ trait TestDriverComponent extends FlatSpec with DriverComponent with Matchers wi
+        workspaceQuery.save(workspace)
+      )
+    }
   }
 
+  class LockedWorkspace() extends TestData {
+    val userOwner = RawlsUser(UserInfo("owner-access", OAuth2BearerToken("token"), 123, "123456789876543212345"))
+    val userWriter = RawlsUser(UserInfo("writer-access", OAuth2BearerToken("token"), 123, "123456789876543212346"))
+    val userReader = RawlsUser(UserInfo("reader-access", OAuth2BearerToken("token"), 123, "123456789876543212347"))
+    val wsName = WorkspaceName("myNamespace", "myWorkspace")
+    val ownerGroup = makeRawlsGroup(s"${wsName} OWNER", Set(userOwner), Set.empty)
+    val writerGroup = makeRawlsGroup(s"${wsName} WRITER", Set(userWriter), Set.empty)
+    val readerGroup = makeRawlsGroup(s"${wsName} READER", Set(userReader), Set.empty)
+
+    val workspace = Workspace(wsName.namespace, wsName.name, UUID.randomUUID().toString, "aBucket", DateTime.now, DateTime.now, "testUser", Map.empty, Map(
+      WorkspaceAccessLevels.Owner -> ownerGroup,
+      WorkspaceAccessLevels.Write -> writerGroup,
+      WorkspaceAccessLevels.Read -> readerGroup), isLocked = true )
+
+    override def save() = {
+      DBIO.seq (
+        rawlsUserQuery.save(userOwner),
+        rawlsUserQuery.save(userWriter),
+        rawlsUserQuery.save(userReader),
+        rawlsGroupQuery.save(ownerGroup),
+        rawlsGroupQuery.save(writerGroup),
+        rawlsGroupQuery.save(readerGroup),
+        workspaceQuery.save(workspace)
+      )
+    }
+  }
+
+  class DefaultTestData() extends TestData {
+    // setup workspace objects
+    val userOwner = RawlsUser(UserInfo("owner-access", OAuth2BearerToken("token"), 123, "123456789876543212345"))
+    val userWriter = RawlsUser(UserInfo("writer-access", OAuth2BearerToken("token"), 123, "123456789876543212346"))
+    val userReader = RawlsUser(UserInfo("reader-access", OAuth2BearerToken("token"), 123, "123456789876543212347"))
+    val wsName = WorkspaceName("myNamespace", "myWorkspace")
+    val ownerGroup = makeRawlsGroup(s"${wsName} OWNER", Set(userOwner), Set.empty)
+    val writerGroup = makeRawlsGroup(s"${wsName} WRITER", Set(userWriter), Set.empty)
+    val readerGroup = makeRawlsGroup(s"${wsName} READER", Set(userReader), Set.empty)
+
+    val billingProject = RawlsBillingProject(RawlsBillingProjectName(wsName.namespace), Set(RawlsUser(userInfo)), "testBucketUrl")
+
+    val wsAttrs = Map(
+      "string" -> AttributeString("yep, it's a string"),
+      "number" -> AttributeNumber(10),
+      "empty" -> AttributeEmptyList,
+      "values" -> AttributeValueList(Seq(AttributeString("another string"), AttributeString("true")))
+    )
+
+    val workspaceNoGroups = Workspace(wsName.namespace, wsName.name + "3", UUID.randomUUID().toString, "aBucket2", DateTime.now, DateTime.now, "testUser", wsAttrs, Map.empty)
+
+    val workspace = Workspace(wsName.namespace, wsName.name, UUID.randomUUID().toString, "aBucket", DateTime.now, DateTime.now, "testUser", wsAttrs, Map(
```

Comment:
I am not inclined to change this at the moment. It is not used by any tests yet and will make porting the tests that will use it difficult.